packages/client/src/components/app/CodeScanner.svelte

```diff
@@ -0,0 +1,235 @@
+  import { Html5Qrcode } from "html5-qrcode"
+
+  export let value
+  export let disabled = false
+  export let allowManualEntry = false
+  export let scanButtonText = "Scan Code"
+
+  import { createEventDispatcher } from "svelte"
+  const dispatch = createEventDispatcher()
+
+  let videoEle
+  let camModal
+  let manualMode = false
+  let cameraEnabled
+  let cameraStarted = false
+  let html5QrCode
+  let cameraSetting = { facingMode: "environment" }
+  let cameraConfig = {
+    fps: 25,
+    qrbox: { width: 250, height: 250 },
+  }
+  const onScanSuccess = decodedText => {
+    if (value != decodedText) {
+      dispatch("change", decodedText)
+    }
+  }
+
+  const initReader = async () => {
+    if (html5QrCode) {
+      html5QrCode.stop()
+    }
+    html5QrCode = new Html5Qrcode("reader")
+    return new Promise(resolve => {
+      html5QrCode
+        .start(cameraSetting, cameraConfig, onScanSuccess)
+        .then(() => {
+          resolve({ initialised: true })
+        })
+        .catch(err => {
+          console.log("There was a problem scanning the image", err)
+          resolve({ initialised: false })
+        })
+    })
+  }
+
+  const checkCamera = async () => {
+    return new Promise(resolve => {
+      Html5Qrcode.getCameras()
+        .then(devices => {
+          if (devices && devices.length) {
+            resolve({ enabled: true })
+          }
+        })
+        .catch(e => {
+          console.error(e)
+          resolve({ enabled: false })
+        })
+    })
+  }
+
+  const start = async () => {
+    const status = await initReader()
+    cameraStarted = status.initialised
+  }
+
+  $: if (cameraEnabled && videoEle && !cameraStarted) {
+    start()
+  }
+
+  const showReaderModal = async () => {
+    camModal.show()
+    const camStatus = await checkCamera()
+    cameraEnabled = camStatus.enabled
+  }
+
+  const hideReaderModal = async () => {
+    cameraEnabled = undefined
+    cameraStarted = false
+    if (html5QrCode) {
+      await html5QrCode.stop()
+      html5QrCode = undefined
+    }
+    camModal.hide()
+  }
+</script>
+
+<div class="scanner-video-wrapper">
+  {#if value && !manualMode}
+    <div class="scanner-value field-display">
+      <StatusLight positive />
+      {value}
+    </div>
+  {/if}
+
+  {#if allowManualEntry && manualMode}
+    <div class="manual-input">
+      <Input
+        bind:value
+        on:change={() => {
+          dispatch("change", value)
+        }}
+      />
+    </div>
+  {/if}
+
+  {#if value}
+    <ActionButton
+      on:click={() => {
+        dispatch("change", "")
+      }}
+      {disabled}
+    >
+      Clear
+    </ActionButton>
+  {:else}
+    <ActionButton
+      icon="Camera"
+      on:click={() => {
+        showReaderModal()
+      }}
+      {disabled}
+    >
+      {scanButtonText}
+    </ActionButton>
+  {/if}
+</div>
+
+<div class="modal-wrap">
+  <Modal bind:this={camModal} on:hide={hideReaderModal}>
+    <ModalContent
+      title={scanButtonText}
```

Comment:
`scanButtonText` will default to "Scan code" if the value is undefined, but if the user has entered a value in the setting field and then cleared it, the text will always be undefined as it does not account for empty strings. 

2 ways to solve this would be:
`$: scanButtonText = scanButtonText || "Scan code"` 
or every time you use `scanButtonText`, do
`scanButtonText || "Scan code`